doc/xml/parsing.md

```diff
@@ -0,0 +1,504 @@
+xml = '<root><foo/><!-- Comment text. --><bar/></root>'
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(Element: {
+      name = "root",
+      children = [
+        #(Element: { name = "foo" }),
+        #(Comment " Comment text. "),
+        #(Element: { name = "bar" })]
+      })]
+  })
+```
+
+### Tag Attributes
+
+\Nokogiri parses an [XML tag attribute](https://www.w3.org/TR/REC-xml/#NT-Attribute)
+into a Nokogiri::XML::Attr object:
+
+```
+xml = '<root foo="0" bar="1"/>'
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(Element: {
+      name = "root",
+      attribute_nodes = [
+        #(Attr: { name = "foo", value = "0" }),
+        #(Attr: { name = "bar", value = "1" })]
+      })]
+  })
+```
+
+### Element Type Declarations
+
+\Nokogiri parses an [XML element type declaration](https://www.w3.org/TR/REC-xml/#elemdecls)
+into a Nokogiri::XML::ElementDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ELEMENT note (to,from,heading,body)>
+<!ELEMENT to (#PCDATA)>
+<!ELEMENT from (#PCDATA)>
+<!ELEMENT heading (#PCDATA)>
+<!ELEMENT body (#PCDATA)>
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [
+        #(ElementDecl: { "<!ELEMENT note (to , from , heading , body)>\n" }),
+        #(ElementDecl: { "<!ELEMENT to (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT from (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT heading (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT body (#PCDATA)>\n" })]
+      })]
+  })
+  ```
+
+### Attribute-List Declarations
+
+\Nokogiri parses an [XML attribute-list declaration](https://www.w3.org/TR/REC-xml/#attdecls)
+into a Nokogiri::XML::AttributeDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ELEMENT payment (#PCDATA)>
+<!ATTLIST payment type CDATA "check">
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [
+        #(ElementDecl: { "<!ELEMENT payment (#PCDATA)>\n" }),
+        #(AttributeDecl: { "<!ATTLIST payment type CDATA \"check\">\n" })]
+      })]
+  })
+```
+
+### Conditional Sections
+
+\Nokogiri parses an [XML conditional section](https://www.w3.org/TR/REC-xml/#sec-condition-sect)
+into a Nokogiri::XML::EntityDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ENTITY % draft 'INCLUDE' >
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [ #(EntityDecl: { "<!ENTITY % draft \"INCLUDE\">\n" })]
+      })]
+  })
+```
+
+### Character References
+
+\Nokogiri parses an [XML character reference](https://www.w3.org/TR/REC-xml/#sec-references)
+(such as <tt>&#9792;</tt>)
+and replaces it with a character such as (<tt>'♀'</tt>):
```

Comment:
You'll need to escape the string in the `<tt>` tags because a web browser will render it as a character (and not the character reference):

![image](https://github.com/user-attachments/assets/56d2e756-8501-47f1-9826-b3fff3a7b252)

Reply:
Fixed (but not in the way I first thought, with a prefixed `'\'`).